eth/backend.go

```diff
@@ -40,13 +41,18 @@ type Backend interface {
 	ethereum.LogFilterer
 	ethereum.ChainReader
 	ChainID(ctx context.Context) (*big.Int, error)
+	MaxGasPrice() *big.Int
+	SetMaxGasPrice(gp *big.Int)
 }
 
 type backend struct {
 	*ethclient.Client
 	abiMap       map[string]*abi.ABI
 	nonceManager *NonceManager
 	signer       types.Signer
+
+	sync.Mutex
```

Comment:
Should we use a RWMutex here since the # of reads of the maxGasPrice will likely far exceed the # of writes in practice?